autoprecompiles/src/lib.rs

```diff
@@ -294,18 +293,35 @@ pub const EXECUTION_BUS_ID: u64 = 0;
 pub const MEMORY_BUS_ID: u64 = 1;
 pub const PC_LOOKUP_BUS_ID: u64 = 2;
 
-pub fn build<T: FieldElement>(
+/// A configuration of a VM in which execution is happening.
+pub struct VmConfig<'a, T: FieldElement, B> {
+    /// Maps an opcode to its kind.
+    pub instruction_kind: BTreeMap<usize, InstructionKind>,
```

Comment:
Actually I think we don't really use this inside apc right now. Everything branch related is handled by `customize` afair